core/src/main/kotlin/maru/consensus/ForkSpec.kt

```diff
@@ -26,27 +27,27 @@ class ForksSchedule(
   val chainId: UInt,
   forks: Collection<ForkSpec>,
 ) {
+  private val log = LogManager.getLogger(this.javaClass)
   private val forks: NavigableSet<ForkSpec> =
     run {
       val newForks =
         TreeSet(
-          Comparator.comparing(ForkSpec::timestampSeconds).reversed(),
+          Comparator
+            .comparing(ForkSpec::timestampSeconds)
+            .thenComparing { forks.indexOf(it) }
```

Comment:
Why is this needed? If we have 2 forks with the timestamp, it should be an error. We can even put a validation for it